python/tvm/driver/tvmc/runner.py

```diff
@@ -0,0 +1,450 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+Provides support to run compiled networks both locally and remotely.
+"""
+import json
+import logging
+import os
+import tarfile
+import tempfile
+
+import numpy as np
+import tvm
+from tvm import rpc
+from tvm.autotvm.measure import request_remote
+from tvm.contrib import graph_runtime as runtime
+from tvm.contrib.debugger import debug_runtime
+
+from . import common
+from .common import TVMCException
+from .main import register_parser
+
+
+# pylint: disable=invalid-name
+logger = logging.getLogger("TVMC")
+
+
+@register_parser
+def add_run_parser(subparsers):
+    """ Include parser for 'run' subcommand """
+
+    parser = subparsers.add_parser("run", help="run a compiled module")
+    parser.set_defaults(func=drive_run)
+
+    # TODO --device needs to be extended and tested to support other targets,
+    #      like 'cl', 'webgpu', etc (@leandron)
+    parser.add_argument(
+        "--device",
+        choices=["cpu", "gpu"],
+        default="cpu",
+        help="target device to run the compiled module",
+    )
```

Comment:
Why not just `target`? Suppose users have use `compile` to build a model, then it should be straightforward to use the same target here.

Reply:
I took some time thinking about this option, and playing with it on my machine. I concluded that users might get confused and have expectations if we use `—target` terminology here, as we have it in other subcommands.

This is why I decided to go with the terminology used by the underlying APIs, and call it device instead. What do you think?

Reply:
Hmm...that's my intention to have a consistent subcommend for the same semantic. Could you elaborate what might be confusing in the case of using `--target` here?

Reply:
My understanding of the difference between `target` and `device` is:
* `--target` from both `tvmc compile` and `tvmc tune` will plug into the `Target` API, i.e. `tvm.target.Target(target)` and accepts things like `llvm ...` or ` cuda...` or `c ...`
* `--device`, which I think only belongs to `tvmc run` is connected to a `session`, as in `tvm.rpc.client.RPCSession.*`, where * can be `.cpu()`, `.gpu()`, `.cl()`, `.webgpu()`, etc.

Again, my understanding is the it doesn't make sense `tvmc compile --target cpu` or `tvmc run --target llvm`. This is where I think expectations from an unified `--target` argument would be misleading.

OTOH, in case you think it makes sense or I'm getting something wrong, please let me know.

Reply:
I see, not sure it's because I used TVM all the time so it looks fine to me lol
Let me tag others for more data @zhiics @masahi @tqchen @junrushao1994

Reply:
I agree that target and device are different thing, one is a compile time concept(what to generate), another is runtime context(what GPU to run)

Reply:
+1 for keeping these separate